src/plugins/matrixops/flatten.cpp

```diff
@@ -0,0 +1,313 @@
+            const matrix_column_iterator<decltype(a)> a_end(a, a.columns());
+
+            auto d = result.data();
+
+            for (auto it = a_begin; it != a_end; ++it)
+            {
+                d = std::copy(it->begin(), it->end(), d);
+            }
+        }
+
+        return primitive_argument_type{std::move(result)};
+    }
+
+    primitive_argument_type flatten::flatten2d(
+        primitive_argument_type&& arg, std::string order) const
+    {
+        switch (extract_common_type(arg))
+        {
+        case node_data_type_bool:
+            return flatten2d(
+                extract_boolean_value_strict(std::move(arg), name_, codename_),
+                order);
+
+        case node_data_type_int64:
+            return flatten2d(
+                extract_integer_value_strict(std::move(arg), name_, codename_),
+                order);
+
+        case node_data_type_double:
+            return flatten2d(
+                extract_numeric_value_strict(std::move(arg), name_, codename_),
+                order);
+
+        case node_data_type_unknown:
+            return flatten2d(
+                extract_numeric_value(std::move(arg), name_, codename_), order);
+
+        default:
+            break;
+        }
+
+        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+            "phylanx::execution_tree::primitives::flatten::flatten2d",
+            generate_error_message(
+                "the arange primitive requires for all arguments to "
+                "be numeric data types"));
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    hpx::future<primitive_argument_type> flatten::eval(
+        primitive_arguments_type const& operands,
+        primitive_arguments_type const& args, eval_context ctx) const
+    {
+        if (operands.empty() || operands.size() > 2)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "flatten::eval",
+                generate_error_message("the flatten primitive requires exactly "
+                                       "one or two operands"));
+        }
+
+        for (auto const& i : operands)
+        {
+            if (!valid(i))
+            {
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "flatten::eval",
+                    generate_error_message(
+                        "the flatten primitive requires that the "
+                        "arguments given by the operands array are valid"));
+            }
+        }
+
+        auto this_ = this->shared_from_this();
+        if (operands.size() == 1)
+        {
+            return hpx::dataflow(hpx::launch::sync,
+                hpx::util::unwrapping(
+                    [this_ = std::move(this_)](primitive_argument_type&& arg)
+                        -> primitive_argument_type {
+                        std::string order = "C";
+                        if (order != "C" && order != "F")
```

Comment:
`order` will always be equal to `'C'`, is this an oversight, will there be more added later?

Reply:
When order is not specified it should be 'C', I copied from the part when operands.size() !=1 but forgot to change this part sorry.